### ISSUE TITLE
Wrong label's "for" attribute

### ISSUE THREAD

**Issue by billmn** [fuel/core#751]
I'm using fieldset class (but I think this is a problem of Form class) ... and I notice that labels have a wrong "for" attribute, the auto_id_prefix is double.

<pre>
&lt;label for="form_form_code" class="control-label">Code&lt;/label>
&lt;input type="text" value="" name="code" id="form_code" required="required">
</pre>

**Comment by Mitchell64** [comment]
I noticed the same thing and ended up removing the auto_id_prefix by setting it to ''.

I also noticed a couple of other issues with the form output that I documented in Issue #758.

**Comment by sergiigrebeniuk** [comment]
Hi Davide.
I can't reproduce the issue.
This code:

``` ruby
$form = Fieldset::forge();
$form->add('code', 'Code', array('type' => 'text'));
echo $form;
```

outputs:

``` html
<form action="http://fuelphp.tld/index.php/issues/751" accept-charset="utf-8" method="post">

<table>
<tr>
    <td class=""><label for="form_code">Code</label></td>
    <td class=""><input type="text" id="form_code" name="code" value="" /> </td>
</tr>

</table>
</form>
```

Could you give us some examples?

**Comment by sergiigrebeniuk** [comment]
Actually I updated version and managed to reproduce this.

**Comment by krtek4** [comment]
This issue is still present in the 1.1/develop branch as of today.

BTW, I was wondering what is the "meaning" of the $id parameter of the method label in the Form_Instance class. Since the 'id' key is unset from the attributes array, the label won't have any html id after the generation, and if it represents the id of the element the label is for, then there should'nt be any other treatment done on the variable.

Not sur I'm clear enough, so I'll provide some usecases :

1° 
- auto_id = false in the config
- I pass the value 'test' as the $id
  ==> the HTML label won't have any for attribute, $id isn't used at all
  IHMO, the for attribute should be 'test'

2° 
- auto_id = false in the config
- I pass the value 'test' as the $id
- $attributes['for'] is set as 'my_input'
  ==> the HTML label will have 'my_input' as its for attribute, $id isn't used at all
  I don't know what is the meaning of the $id parameter in this case

3° 
- auto_id = true in the config
- I pass the value 'test' as the $id
  ==> the HTML label will have 'form_test' as its for attribute (or auto_id_prefix insteand of 'form_' depending on the config
  IHMO, If $id is supposed to be the name of the element referenced by the label, then the for attribute should be test.

4° 
- auto_id = true in the config
- I pass the value 'test' as the $id
- $attributes['for'] is set as 'my_input'
  ==> the HTML label will have 'my_input' as its for attribute, $id isn't used at all
  As in 2°, what should be the meaning of the $id parameter ?

5° 
- auto_id = true in the config
- nothing passed for the $id parameter (ie null)
  ==> the HTML label will have 'form_' as its for attribute (or auto_id_prefix insteand of 'form_' depending on the config
  IHMO, this is cleary a bug

I hope my point of view is clearer now. IHMO, the $id parameter should be renamed $for or completly deleted and it should be used as is since we don't have any information which can be used to compute the name of the referenced element.

**Comment by jschreuder** [comment]
Closed, will be handled by issue #839.

**Comment by Savageman** [comment]
I also have this problem. The prefix gets prepended twice to the for attribute of the label.

`<label for="form_form_whatever">` and `<input id="form_whatever">`
